magento/pwa-studio
# fix: remove unintentionally spammy loadEnvironment warnings

## Description

Simplify the breaking change warnings in `loadEnvironment`, and add a fail-safe to make sure they don't log forever.

Changes:

- Remove `defaultChanged` and `exampleChanged` warning types, because they're not useful and I was wrong to create them. When a default changes, you either don't care or you've overridden it. When an example changes, you really never care.
- Remove the old warnings about changes from many releases ago.

Additions:

- Add a required "dateAdded" and an optional "warnForDays" property to change entries. They will now only log for an interval of "warnForDays", up to 180 days.
- Add a test for `envVarDefinitions` themselves, to ensure they're compliant. 

## Related Issue
Fixes [PWA-502](https://jira.corp.magento.com/browse/PWA-502)
Fixes underlying problem from #2308 

## Acceptance 
### Verification Stakeholders
@brendanfalkowski @revanth0212 
### Specification



### Verification Steps


1. Check out the branch.
2. Run a build. Observe that the CHECKOUT_BRAINTREE_TOKEN is gone.
3. Examine the new tests. Break them experimentally if you want to see what's going on. Or, add new change entries and see what happens.


## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
